File: python/242. Valid Anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        freq = {}

        for i in range(len(s)):
            freq[s[i]] = freq.get(s[i], 0) + 1
            freq[t[i]] = freq.get(t[i], 0) - 1

        for value in freq.values():
            if value != 0:
                return False

        return True

    # def isAnagram(self, s: str, t: str) -> bool:
    #     return sorted(s) == sorted(t)
        

File: java/242. Valid Anagram/Solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        // sort the array and check whether they're eaqual
        if (s.length() != t.length()) return false;
        Map<Character, Integer> frequencies = new HashMap<>();

        for (int i = 0; i < s.length(); i ++) {
            frequencies.put(s.charAt(i), frequencies.getOrDefault(s.charAt(i), 0) + 1);
            frequencies.put(t.charAt(i), frequencies.getOrDefault(t.charAt(i), 0) - 1);
        }

        for (Integer value : frequencies.values()) {
            if (value != 0) return false;
        }

        return true;

    }
}